extension/src/editors/form-editor/webview-communication.ts

```diff
@@ -0,0 +1,26 @@
+import * as vscode from 'vscode';
+import { Messenger } from 'vscode-messenger';
+import { WebSocketForwarder } from '../websocket-forwarder';
+import { DisposableCollection } from '@eclipse-glsp/vscode-integration';
+import { NotificationType } from 'vscode-messenger-common';
+
+const WebviewReadyNotification: NotificationType<void> = { method: 'ready' };
+const InitializeConnectionRequest: NotificationType<{ file: string }> = { method: 'initializeConnection' };
+const FormWebSocketMessage: NotificationType<unknown> = { method: 'formWebSocketMessage' };
+
+export const setupCommunication = (messenger: Messenger, webviewPanel: vscode.WebviewPanel, document: vscode.TextDocument) => {
+  const webSocketAddress = process.env.WEB_SOCKET_ADDRESS;
+  if (!webSocketAddress) {
```

Comment:
I would move that web socket address resolution into a command, as we do it twice now.

Reply:
I suggest providing this websocket address over the context instead of the environment and reducing these commands as they are more for communication between different extensions than for internal calls IMO. (As discussed on the phone :))